pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -62,6 +62,17 @@ var splittableRangeVectorAggregators = map[string]bool{
 	sumOverTime:   true,
 }
 
+// cannotDoubleCountBoundaries is the list of functions that cannot double count the boundaries points when being split by range.
+// Consider a timeline with samples at 1s,2s,3s,4s,5s...
+// Prometheus ranges are closed on both ends, so range [2s,4s] includes samples at 2s and 4s.
+// If we split that into [2s, 3s] and [3s, 4s], then we will include the sample at 3s in both ranges.
+// This cannot be done for some sample-aggregating functions like `count_over_time`: we would count 4 samples (2s,3s,3s,4s) after splitting.
+// However, for other functions, like `rate`, it is important to have both ends includes, as the rate is (start-end)/range.
+var cannotDoubleCountBoundaries = map[string]bool{
```

Comment:
Great comment 👍 and already understand it. 
It does not affect `max_over_time` and `min_over_time` because the same value is considered 2 times and it either is the max/min or not.

Small nit: maybe can you also add `avg_over_time` in the comment, since this will be converted in a `sum_over_time / count_over_time)` and should **not** be added to the map

Reply:
I think you're right, `avg_over_time` also has to be here. But we need to build a testcase for it.

Reply:
Doesn't need to be in the map, because it will be split into `sum_over_time / count_over_time` which are already on the map. But it would be good to a comment explaining that for future reference